qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferPresenter.cpp

```diff
@@ -0,0 +1,257 @@
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ISISEnergyTransferPresenter.h"
+#include "ISISEnergyTransferData.h"
+#include "ISISEnergyTransferModel.h"
+#include "ISISEnergyTransferView.h"
+#include "IndirectSettingsHelper.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidKernel/ConfigService.h"
+
+#include "MantidQtWidgets/Common/UserInputValidator.h"
+
+#include <filesystem>
+
+using namespace Mantid::API;
+using MantidQt::API::BatchAlgorithmRunner;
+
+namespace {
+bool doesExistInADS(std::string const &workspaceName) {
+  return AnalysisDataService::Instance().doesExist(workspaceName);
+}
+} // namespace
+
+namespace MantidQt::CustomInterfaces {
+
+IETPresenter::IETPresenter(IndirectDataReduction *idrUI, QWidget *parent)
+    : IndirectDataReductionTab(idrUI, parent), m_model(std::make_unique<IETModel>()),
+      m_view(std::make_unique<IETView>(this, parent)) {
+
+  setOutputPlotOptionsPresenter(
+      std::make_unique<IndirectPlotOptionsPresenter>(m_view->getPlotOptionsView(), PlotWidget::SpectraContour));
+
+  connect(this, SIGNAL(newInstrumentConfiguration()), this, SLOT(setInstrumentDefault()));
+
+  connect(this, SIGNAL(updateRunButton(bool, std::string const &, QString const &, QString const &)), m_view.get(),
+          SLOT(updateRunButton(bool, std::string const &, QString const &, QString const &)));
+}
+
+IETPresenter::~IETPresenter() = default;
+
+void IETPresenter::setup() {}
+
+bool IETPresenter::validateInstrumentDetails() {
+  auto const instrument = getInstrumentName().toStdString();
+  if (instrument.empty()) {
+    showMessageBox("Please select a valid facility and/or instrument.");
+    return false;
+  }
+
+  QMap<QString, QString> instrumentDetails = getInstrumentDetails();
+  std::set<std::string> keys = {"spectra-min", "spectra-max"};
+  for (const auto &key : keys) {
+    if (!instrumentDetails.contains(QString::fromStdString(key)) ||
+        instrumentDetails[QString::fromStdString(key)].isEmpty()) {
+      showMessageBox(QString::fromStdString("Could not find " + key + " for the " + instrument +
+                                            " instrument. Please select a valid instrument."));
+      return false;
+    }
+  }
+
+  return true;
+}
+
+InstrumentData IETPresenter::getInstrumentData() {
+  auto instrument = getInstrumentName().toStdString();
+  auto analyser = getAnalyserName().toStdString();
+  auto reflection = getReflectionName().toStdString();
+
+  QMap<QString, QString> instrumentDetails = getInstrumentDetails();
+
+  auto specMin = instrumentDetails["spectra-min"].toInt();
+  auto specMax = instrumentDetails["spectra-max"].toInt();
+  auto eFixed = instrumentDetails["Efixed"].toDouble();
+  auto rebinDefault = instrumentDetails["rebin-default"].toStdString();
+  auto cm1ConvertChoice = instrumentDetails["cm-1-convert-choice"] == "true";
+  auto saveNexusChoice = instrumentDetails["save-nexus-choice"] == "true";
```

Comment:
what does  `instrumentDetails["cm-1-convert-choice"]` return? Could we not just do a static cast into a bool?

Also you dont need to assign to these variables and could just send them straight to the object.

Reply:
It returns a boolean for the cm1 unit option in the output section of the GUI. These are the default value for the fields in the GUI.
I intentionally add new variables to avoid long lines because the object has a lot of parameters

Reply:
What is the purpose of the additional `== "true"`? Could we not just have `bool(instrumentDetails["thing"])`?

Reply:
Yes, I could write it that way as it is clearer